cpp/src/arrow/compute/kernels/vector_selection_test.cc

```diff
@@ -1734,5 +1734,240 @@ TEST(TestTake, RandomFixedSizeBinary) {
   TakeRandomTest<FixedSizeBinaryType>::Test(fixed_size_binary(16));
 }
 
+// ----------------------------------------------------------------------
```

Comment:
I think we need to add following test cases
1. 0-lengthed inputs  (to test the early termination code paths)
2. non-zero but all null values on inputs  (to test the early termination code paths)
3. larger test case with more random values (say 1000 or so) --> Because bitmap ops work on 8 elements together.  So, we need to test beyond 8 elements in an array. 
ex: https://github.com/apache/arrow/blob/master/cpp/src/arrow/compute/kernels/scalar_if_else_test.cc#L56
4. also, it would be nicer to have test with non-zero offsets in the inputs (only for array impl)